app/serializers/api/v3/npq_participant_serializer.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "jsonapi/serializer/instrumentation"
+
+module Api
+  module V3
+    class NPQParticipantSerializer
+      include JSONAPI::Serializer
+      include JSONAPI::Serializer::Instrumentation
+
+      set_id :id
+      set_type :'npq-participant'
+
+      attribute :full_name
+
+      attribute(:teacher_reference_number) do |object|
+        object.teacher_profile&.trn
+      end
+
+      attribute(:updated_at) do |object|
+        object.updated_at.rfc3339
+      end
+
+      attribute(:npq_enrolments) do |object, params|
+        scope = object.npq_profiles
+        scope = scope.includes(:npq_course, :npq_application, :participant_identity, :participant_profile_states, schedule: [:cohort])
+
+        if params[:cpd_lead_provider]
+          scope = scope.joins(npq_application: { npq_lead_provider: [:cpd_lead_provider] })
+          scope = scope.where(npq_applications: { npq_lead_providers: { cpd_lead_provider: params[:cpd_lead_provider] } })
+        end
+
+        scope.map do |profile|
+          hash = {
```

Comment:
I think we have n+1s when surfacing this data in the endpoint, could we amend/improve the queries?

Reply:
chatted in person: issue not introduced in this pr but if we could remove the extra queries for v3 when mapping profiles as it takes a long time to render the results